crates/formality-prove/src/prove/prove_wf.rs

```diff
@@ -39,10 +39,33 @@ judgment_fn! {
             (prove_wf(decls, env, assumptions, RigidTy { name: RigidName::ScalarId(_), parameters }) => c)
         )
 
+        (
+            (for_all(&decls, &env, &assumptions, &parameters, &prove_wf) => c)
+            --- ("integers and booleans")
```

Comment:
```suggestion
            --- ("ADT")
```

Reply:
we should be checking the where-clauses on the ADT declaration, but let's leave that for a follow-up PR